app/models/ErrorState.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+sealed trait ErrorState {
+  val reason: String
+}
+
+case class SubmissionSuccess(departure: Departure)
+
+sealed trait InternalError extends ErrorState
+
+sealed trait ExternalError extends ErrorState
+
+case class DepartureNotFound(reason: String)         extends ExternalError
+case class InvalidMessageType(reason: String)        extends ExternalError
+case class TransitionError(reason: String)           extends ExternalError
+case class XMLMRNError(reason: String)               extends ExternalError
+case class SubmissionFailureRejected(reason: String) extends ExternalError
+
+case object SubmissionFailureExternal extends ExternalError {
+  override val reason: String = ""
+}
+
+case object SubmissionFailureInternal extends InternalError {
```

Comment:
Perhaps we could add the exception to both flavours of `SubmissionFailureInternal` so that it can be logged in the orchestrator too? 
Although I guess we would want to be careful that things don't get logged multiple times